### PR TITLE
Refactoring/52 data conv utility

### LINKED ISSUE #52: Make data conversion utility for the secret store commonly accessible
There is a useful utility - `_optional_str_to_bool` in the connections.py. Let's make it accessible for all modules in the notebook-connector.

### PR DESCRIPTION
**Pull request by ahsimb**
closes #52

### CODE REVIEW

File: exasol/connections.py [hunk 1]
```diff
@@ -37,7 +26,7 @@ def _extract_ssl_options(conf: Secrets) -> dict:
     sslopt: dict[str, object] = {}
 
     # Is server certificate validation required?
-    certificate_validation = _optional_str_to_bool(conf.get("CERTIFICATE_VALIDATION"))
+    certificate_validation = optional_str_to_bool(conf.get("CERTIFICATE_VALIDATION"))
```
Comment (Nicoretti) [inline exasol/connections.py:29]:
```suggestion
    requires_certificate_validation = optional_str_to_bool(conf.get("CERTIFICATE_VALIDATION"))
```

File: exasol/utils.py [hunk 2]
```diff
@@ -17,3 +18,22 @@ def upward_file_search(file_name: str) -> str:
...
+    """
+    if value is None:
+        return None
+    value_l = value.lower()
+    if value_l in ["y", "yes", "true"]:
+        return True
+    elif value_l in ["n", "no", "false"]:
+        return False
+    else:
+        raise ValueError("Invalid boolean value " + value)
```
Comment (Nicoretti) [inline exasol/utils.py:39]:
```suggestion
def optional_str_to_bool(value: Optional[str]) -> Optional[bool]:
    """
    Converts an optional string value to an optional boolean.
    None, '' => None.
    Case-insensitive "y", "yes", "true" => True
    Case-insensitive "n", "no", "false" => False
    Any other value cause a ValueError exception.
    """
    if not value:
        return None

    mapping = {
        'y': True,
        'yes': True,
        'true': True,
        'n': False,
        'no': False,
        'false': False,
    }
    key = value.lower()

    try:
        result = mapping[key]
    except KeyError as ex:
        raise ValueError("Invalid boolean value " + value) from ex

    return result
```

Comment (Nicoretti) [reply exasol/utils.py:39]: Maybe with fewer conditions, it will be easier to read and extend? (suggestion above)

**Review by Nicoretti — APPROVED** [review]

File: exasol/utils.py (hunk 2 above)
Comment (ahsimb) [reply exasol/utils.py:39]: Thanks, yes, looks nicer this way. Also, there is a small bug in my code: an empty string will raise an exception, contrary to what the docstring says.

**Review by Nicoretti — APPROVED** [review]

**Pull request merged**